instrumentation/opentelemetry-instrumentation-boto3sqs/src/opentelemetry/instrumentation/boto3sqs/__init__.py

```diff
@@ -0,0 +1,430 @@
+        `__iter__` functions and the `__getitem__` functions to set the span context of the addressed message. Since
+        the return value from an `SQS.ReceiveMessage` returns a builtin list, we cannot wrap it and change all of the
+        calls for `list.__iter__` and `list.__getitem__` - therefore we use ContextableList. It is bound to the
+        received_messages_spans dict
+        """
+
+        def __getitem__(
+            self, *args: List[Any], **kwargs: Dict[str, Any]
+        ) -> Any:
+            retval = super(
+                Boto3SQSInstrumentor.ContextableList, self
+            ).__getitem__(*args, **kwargs)
+            if not isinstance(retval, dict):
+                return retval
+            if not (receipt_handle := retval.get("ReceiptHandle", None)):
+                return retval
+            if not (
+                started_span := Boto3SQSInstrumentor.received_messages_spans.get(
+                    receipt_handle, None
+                )
+            ):
+                return retval
+            if Boto3SQSInstrumentor.current_context_token:
+                context.detach(Boto3SQSInstrumentor.current_context_token)
+            Boto3SQSInstrumentor.current_context_token = context.attach(
+                trace.set_span_in_context(started_span)
+            )
+            Boto3SQSInstrumentor.current_span_related_to_token = started_span
+            return retval
+
+        def __iter__(self) -> Generator:
+            index = 0
+            while index < len(self):
+                yield self[index]
+                index = index + 1
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    @staticmethod
+    def _enrich_span(
+        span: Span,
+        queue_name: str,
+        conversation_id: Optional[str] = None,
+        operation: Optional[MessagingOperationValues] = None,
+        message_id: Optional[str] = None,
+    ) -> None:
+        if not span.is_recording():
+            return
+        span.set_attribute(SpanAttributes.MESSAGING_SYSTEM, "aws.sqs")
+        span.set_attribute(SpanAttributes.MESSAGING_DESTINATION, queue_name)
+        span.set_attribute(
+            SpanAttributes.MESSAGING_DESTINATION_KIND,
+            MessagingDestinationKindValues.QUEUE.value,
+        )
+        if operation:
+            span.set_attribute(
+                SpanAttributes.MESSAGING_OPERATION, operation.value
+            )
+        else:
+            span.set_attribute(SpanAttributes.MESSAGING_TEMP_DESTINATION, True)
+        if conversation_id:
+            span.set_attribute(
+                SpanAttributes.MESSAGING_CONVERSATION_ID, conversation_id
+            )
+        if message_id:
+            span.set_attribute(SpanAttributes.MESSAGING_MESSAGE_ID, message_id)
+
+    @staticmethod
+    def _safe_end_processing_span(receipt_handle: str) -> None:
+        if started_span := Boto3SQSInstrumentor.received_messages_spans.pop(
```

Comment:
same comment as before for walrus operator (':=')

Reply:
Done